server/api/files.ts

```diff
@@ -1,26 +1,37 @@
 import { FileObject } from "@supabase/storage-js";
-import { FileObjectKeys } from "~~/types/FileObjectKeys";
+import { FileObjectKeys } from "@/types/FileObjectKeys";
+import { SortOrder } from "@/types/SortOrder";
+import { QueryParams } from "@/types/QueryParams";
 import { serverSupabaseClient } from "#supabase/server";
 
 export default defineEventHandler(async (event) => {
   const client = serverSupabaseClient(event);
 
+  let files: FileObject[];
+
   // fetch data from supabase
   const { data, error } = await client.storage.from("files").list("public", {
     limit: 100,
     offset: 0,
   });
   if (error) throw error;
 
+  files = data;
+
   const query = getQuery(event);
 
-  const { key, order } = query as {
+  const { key, order, name, types, minSize, maxSize, dates } = query as {
     key: FileObjectKeys;
-    order: "asc" | "desc";
+    order: SortOrder;
+    name: string;
+    types: string;
+    minSize: number;
+    maxSize: number;
+    dates: string;
   };
```

Comment:
Here I wanted to use newly created type QueryParams, but I don't know how to do it. 
const query = getQuery(event); - this is from h3: https://www.jsdocs.io/package/h3#getQuery
Tried couple of combination but still it wants me to list all the keys and pass its type. Maybe you have some idea?

Reply:
it's not intuitive TS behavior where it doesn't allow you to just assert type 🤷 to resolve it you need to assign this variable as `unknown` first 😶 
`query as unknown as QueryParams;`

Reply:
Thank you :), I have never seen anything like this before :)